contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+
+    modifier onlyNonZeroAddress(address _address) {
+        require(_address != address(0), "Must input non-zero address");
+        _;
+    }
+
+    modifier onlyNonEmptyString(string memory _string) {
+        require(bytes(_string).length != 0, "Must input non-empty string");
+        _;
+    }
+
+    modifier onlyAdminACL(bytes4 _selector) {
+        require(
+            adminACLAllowed(msg.sender, address(this), _selector),
+            "Only Admin ACL allowed"
+        );
+        _;
+    }
+
+    modifier onlyExistingDependencyType(bytes32 _dependencyTypeId) {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type does not exist"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract.
+     * @param _adminACLContract Address of admin access control contract, to be
+     * set as contract owner.
+     */
+    constructor(address _adminACLContract) {
+        // set AdminACL management contract as owner
+        _transferOwnership(_adminACLContract);
+    }
+
+    /**
+     * @notice Adds a new dependency type.
+     * @param _dependencyTypeId Name of dependency type (i.e. "type@version")
+     * @param _preferredCDN Preferred CDN for dependency type.
+     * @param _preferredRepository Preferred repository for dependency type.
+     */
+    function addDependencyType(
+        bytes32 _dependencyTypeId,
+        string memory _preferredCDN,
+        string memory _preferredRepository,
+        string memory _projectWebsite
+    ) external onlyAdminACL(this.addDependencyType.selector) {
+        require(
+            !_dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type already exists"
+        );
+        require(
+            _dependencyTypeId.containsExactCharacterQty(
+                AT_CHARACTER_CODE,
+                uint8(1)
+            ),
+            "must contain exactly one @"
+        );
+
+        _dependencyTypes.add(_dependencyTypeId);
+        dependencyTypeInfo[_dependencyTypeId].preferredCDN = _preferredCDN;
+        dependencyTypeInfo[_dependencyTypeId]
+            .preferredRepository = _preferredRepository;
+        dependencyTypeInfo[_dependencyTypeId].projectWebsite = _projectWebsite;
+
+        emit DependencyTypeAdded(
+            _dependencyTypeId,
+            _preferredCDN,
+            _preferredRepository,
+            _projectWebsite
+        );
+    }
+
+    /**
+     * @notice Removes a new dependency type.
+     * @param _dependencyTypeId Name of dependency type (i.e. "type@version")
+     */
+    function removeDependencyType(bytes32 _dependencyTypeId)
+        external
+        onlyAdminACL(this.removeDependencyType.selector)
+    {
+        require(
+            _dependencyTypes.contains(_dependencyTypeId),
+            "Dependency type does not exist"
+        );
+
+        _dependencyTypes.remove(_dependencyTypeId);
+        delete dependencyTypeInfo[_dependencyTypeId];
```

Comment:
> I think we should probably just require additionalCDNCount, additionalRepositoryCount, and scriptCount to all be zero when this function is called, which would ensure we have no stale values in the mappings.

This approach SGTM, given that removal of dependency types should be a generally infrequent occurrence and, as such, it being a somewhat manual process (multiple contract calls) seems reasonable to me.

Reply:
Any reason to not clear things in this function vs forcing it to be done manually?

Reply:
I think clearing here would be fine, since we wouldn't expect to hit the gas limit. Since the operation is likely rare, keeping on-chain code simple would be the only reason to not do that in this function. A judgement call for sure!